### ISSUE TITLE
no-mixed-spaces-and-tabs doesn't warn for comments with tabs

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#2256]
The below code doesn't give out any warnings for `no-mixed-spaces-and-tabs` rule.

```.js
function runJasmineTests(activeCodeCoverageState, callback){
    var configObj = {};
	// hi
    var b;
}
```

The line 3 (`// hi`) actually has a tab and all other lines have spaces.

Eslint: 0.18.0
Parser: Default

Is this the expected behavior when it comes to comments?

**Comment by gyandeeps** [comment]
Further investigating I found that this rule only check on line bases, what mean is on a line you should not have a mix of spaces and tabs both but I dont this it checks for the same things from a file perspective. 
As in if I follow spaces only and then i should not have any tabs inside the file no matter which line your are on.
As of now i can have one line indented with spaces all the way and the following line indented with tabs all the way, and this rule will not complain. Which is kinda what I am looking for.
Isn't the name misleading upfront as someone would think that its going to apply this rule on a file bases and not line bases.

**Comment by nzakas** [comment]
The rule is behaving as expected, it's job is to tell you when you're using spaces and tabs one after the other on the same line.

The behavior you want is implemented in the indent rule.
